lib/collection/src/wal_delta.rs

```diff
@@ -154,10 +176,10 @@ impl RecoverableWal {
 /// If `None` - the remote WAL is already equal, and we don't have to send any records.
 /// If `Err` - no delta can be resolved.
 fn resolve_wal_delta(
+    operations: impl DoubleEndedIterator<Item = (u64, Option<ClockTag>)>,
```

Comment:
This one is potentially dangerous, because we say nothing about the order we expect.